lib/puppetserver/ca/action/list.rb

```diff
@@ -78,42 +87,67 @@ def run(input)
           all_certs = get_all_certs(puppet.settings).select { |cert| filter_names.call(cert) }
           requested, signed, revoked = separate_certs(all_certs)
           missing = certnames - all_certs.map { |cert| cert['name'] }
+          
+          if all && output_format == 'json'
+            grouped_cert = { "requested" => requested,
+                             "signed" => signed, 
+                             "revoked" => revoked,
+                             "missing" => missing }.to_json
+            @logger.inform(grouped_cert)
+            return 0 
```

Comment:
We should try to avoid returning like this from the middle of a large method body.

Reply:
One thing we could try to do to avoid that is create a data structure that has all the info in it and then pass it to the correct printer method (perhaps even wrapping that choice in a method).

Reply:
I ended up moving the logic down into the function `output_cert_by_states` so that I don't have to return there. Currently debugging why I can't print out the json with python pretty print.